### DISCUSSION TITLE
No LOCAL option in external storage app

### DISCUSSION THREAD

**Discussion by peastuti** [nextcloud/all-in-one#1138] [Questions] [answered]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Install nextcloud all-in-one on docker and complete initial configuration
2. follow[ this guide](https://github.com/nextcloud/all-in-one#how-to-allow-the-nextcloud-container-to-access-directories-on-the-host) to enable access to host directories

### Expected behavior 
The LOCAL option should appear

### Actual behavior 
LOCAL option won't appear (see screenshot)
![Screenshot 2022-09-09 at 00 38 42](https://user-images.githubusercontent.com/18187660/189238795-d2669a45-fe1c-42d9-8420-a0955ac485d3.png)

### Host OS 
Raspberry Pi 4 Model B (arm64)

#### Nextcloud AIO version 
24.0.4

**Comment by szaimen** [comment]
Cannot reproduce. Works for me: 
![image](https://user-images.githubusercontent.com/42591237/189306844-df3dd660-2c71-4dd9-8818-6debe83b5fc8.png)

**Comment by szaimen** [comment, marked as answer]
After adding that option, did you restart your containers from the AIO interface in order to apply that change?

**Reply by fleshcrucifix** [reply]
I have the seem problem and I have restarted the containers in every way I can see (portainer, directly through cli) - but @szaimen can you explain how to restart the containers from the AIO interface?  Thanks!

**Reply by szaimen** [reply]
Simply click on `Stop containers` and `Start containers` in the AIO interface :)

**Reply by fleshcrucifix** [reply]
I guess I don't really understand where the AIO interface is.  I have the Nextcloud interface in front of me, the Portainer interface in front of me, and the CLI - but I don't see (beyond executing CLI commands) a way to stop and start containers.
(Thanks for your patience, you help so many people with this, and I am grateful for your guidance!)

**Reply by szaimen** [reply]
Did you install Nextcloud via AIO or in a different way? Usually the AIO interface should be reachable via port 8080.

**Reply by fleshcrucifix** [reply]
I installed via docker the AIO and I am using cloudflared to get to my server and see the interface {attached}.
![Screenshot_20230823_195648](https://github.com/nextcloud/all-in-one/assets/56274694/cbc5dd35-b0db-4b81-bfe2-d68e4dce9c17)

**Reply by szaimen** [reply]
So did you open the AIO interface already? Usuually running on port 8080?

**Reply by fleshcrucifix** [reply]
Hello @szaimen, I feel so silly!  It has taken me days to understand that you are talking about this as the AIO interface:
![Capture](https://github.com/nextcloud/all-in-one/assets/56274694/ca4a206b-185b-4fb1-aa9a-4d345c49231f)

I will try it tonight - thank you very much for your help!

**Comment by peastuti** [comment]
You are right @szaimen, I forgot to restart the containers. Now I configured it correctly, but still, I'm not able to see the files though :(

**Reply by szaimen** [reply]
> Now I configured it correctly, but still, I'm not able to see the files though :(

based on https://github.com/nextcloud/all-in-one#how-to-run-occ-commands, you could try a `sudo docker exec -it nextcloud-aio-nextcloud php occ files:scan --all`

**Reply by peastuti** [reply]
It says permission denied, even though I executed this
![Screenshot 2022-09-09 at 11 28 21](https://user-images.githubusercontent.com/18187660/189318735-e4272779-cc82-4e80-95e7-da11d5cda755.png)


![Screenshot 2022-09-09 at 11 27 35](https://user-images.githubusercontent.com/18187660/189318584-c395b57e-bb1f-4d2a-9180-14d3dfa06bf7.png)

**Reply by szaimen** [reply]
can you run `docker exec -it nextcloud-aio-nextcloud ls -l /mnt/passport`  and post the output here?

**Reply by peastuti** [reply]
Still, permission denied.

![Screenshot 2022-09-09 at 11 31 04](https://user-images.githubusercontent.com/18187660/189319352-79760f8b-a4fc-41dd-aa35-af00367b1180.png)

this is from my docker-compose:
```
        environment:
            - APACHE_PORT=11000
            - APACHE_IP_BINDING=127.0.0.1
            - SKIP_DOMAIN_VALIDATION=true
            - NEXTCLOUD_MOUNT=/mnt/
  ```

**Reply by szaimen** [reply]
okay, what is the output of `sudo ls -la /mnt`

**Reply by peastuti** [reply]
![Screenshot 2022-09-09 at 11 35 23](https://user-images.githubusercontent.com/18187660/189320248-54cee307-4ea5-40bb-ad01-c680a831e84f.png)

**Reply by szaimen** [reply]
Please try again with the a option:

> okay, what is the output of `sudo ls -la /mnt`

**Reply by peastuti** [reply]
Sorry I didn't get that. I tried `sudo ls -a /mnt` but it only lists that two folders

**Reply by szaimen** [reply]
The question is what is the ACL of the /mnt folder. The a option should list permissons of the parent /mnt directory as well.

**Reply by peastuti** [reply]
![Screenshot 2022-09-09 at 11 56 00](https://user-images.githubusercontent.com/18187660/189324280-25620d8b-bf24-40a7-bb31-6aa60b7350df.png)

yes, you're right

**Reply by szaimen** [reply]
Looks good!

Please try `sudo docker exec --user root -it nextcloud-aio-nextcloud ls -la /mnt` and post tje outout here

**Reply by peastuti** [reply]
![Screenshot 2022-09-09 at 12 16 28](https://user-images.githubusercontent.com/18187660/189328031-188f3c44-4735-4dd8-8e7d-ec01af4d17dc.png)

it seems that www-data is missing?

**Reply by szaimen** [reply]
Is this a NTFS drive?

**Reply by peastuti** [reply]
![Screenshot 2022-09-09 at 12 27 58](https://user-images.githubusercontent.com/18187660/189330007-d945296e-4aa5-4612-97ec-d2e2773badcf.png)

**Reply by szaimen** [reply]
all right. then you need to modify your mount options. These should make it work `uid=www-data,gid=www-data,umask=007`

**Reply by peastuti** [reply]
after the scan it worked! Thanks @szaimen :)

**Comment by Boredandinnocent** [comment]
![Screenshot 2024-11-19 021130](https://github.com/user-attachments/assets/c625df38-375d-4158-bc76-2f699655c70a)
I have tried restarting the containers multiple times but still do not get the option. Appreciate all help. Thanks in advance.

**Reply by szaimen** [reply]
See https://github.com/nextcloud/all-in-one#how-to-allow-the-nextcloud-container-to-access-directories-on-the-host

**Comment by aboyington** [comment]
Does anyone have a fix for this on Unraid?

**Reply by szaimen** [reply]
See https://github.com/nextcloud/all-in-one#how-to-allow-the-nextcloud-container-to-access-directories-on-the-host
